File: backend/app/models/content_analysis.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, Text, ForeignKey, DateTime, JSON, Integer, Float
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.database import Base

class ContentAnalysis(Base):
    __tablename__ = "content_analyses"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    lesson_id: Mapped[str] = mapped_column(String(36), ForeignKey("lessons.id", ondelete="CASCADE"), unique=True)
    subject: Mapped[str] = mapped_column(String(100), nullable=True)
    topic: Mapped[str] = mapped_column(String(255), nullable=True)
    language: Mapped[str] = mapped_column(String(10), default="ar")
    estimated_grade: Mapped[str] = mapped_column(String(50), nullable=True)
    difficulty: Mapped[str] = mapped_column(String(50), nullable=True)
    summary: Mapped[str] = mapped_column(Text, nullable=True)
    concepts: Mapped[dict] = mapped_column(JSON, nullable=True)
    learning_objectives: Mapped[list] = mapped_column(JSON, nullable=True)
    important_facts: Mapped[list] = mapped_column(JSON, nullable=True)
    terminology: Mapped[dict] = mapped_column(JSON, nullable=True)
    potential_questions: Mapped[list] = mapped_column(JSON, nullable=True)
    prompt_version: Mapped[str] = mapped_column(String(50), nullable=True)
    input_tokens: Mapped[int] = mapped_column(Integer, nullable=True)
    output_tokens: Mapped[int] = mapped_column(Integer, nullable=True)
    estimated_cost: Mapped[float] = mapped_column(Float, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(timezone.utc))

    lesson = relationship("Lesson", back_populates="analysis")
    concepts_list = relationship("Concept", back_populates="analysis", cascade="all, delete-orphan")
    games = relationship("Game", back_populates="analysis")

File: app/lib/models/content_analysis.dart
import 'package:json_annotation/json_annotation.dart';

part 'content_analysis.g.dart';

@JsonSerializable(fieldRename: FieldRename.snake)
class ContentAnalysis {
  final String id;
  final String lessonId;
  final String? subject;
  final String? topic;
  final String? language;
  final String? estimatedGrade;
  final String? difficulty;
  final String? summary;
  @JsonKey(defaultValue: <String, dynamic>{})
  final Map<String, dynamic> concepts;
  @JsonKey(defaultValue: <dynamic>[])
  final List<dynamic> learningObjectives;
  @JsonKey(defaultValue: <dynamic>[])
  final List<dynamic> importantFacts;
  @JsonKey(defaultValue: <String, dynamic>{})
  final Map<String, dynamic> terminology;
  @JsonKey(defaultValue: <dynamic>[])
  final List<dynamic> potentialQuestions;
  final String? promptVersion;
  final int? inputTokens;
  final int? outputTokens;
  final double? estimatedCost;
  final DateTime createdAt;

  const ContentAnalysis({
    required this.id,
    required this.lessonId,
    this.subject,
    this.topic,
    this.language,
    this.estimatedGrade,
    this.difficulty,
    this.summary,
    this.concepts = const {},
    this.learningObjectives = const [],
    this.importantFacts = const [],
    this.terminology = const {},
    this.potentialQuestions = const [],
    this.promptVersion,
    this.inputTokens,
    this.outputTokens,
    this.estimatedCost,
    required this.createdAt,
  });

  factory ContentAnalysis.fromJson(Map<String, dynamic> json) => _$ContentAnalysisFromJson(json);
  Map<String, dynamic> toJson() => _$ContentAnalysisToJson(this);
}
